includes/email-signup/class-sensei-email-signup-form.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * File containing the class Sensei_Email_Signup_Form.
+ *
+ * @package sensei
+ * @since   2.0.0
+ */
+
+/**
+ * Class for displaying the modal email signup form.
+ *
+ * @class Sensei_Email_Signup_Form
+ */
+class Sensei_Email_Signup_Form {
+	const MC_USER_ID = '7a061a9141b0911d6d9bafe3a';
+	const MC_LIST_ID = '4fa225a515';
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Email_Signup_Form constructor. Prevents other instances from being
+	 * created outside of `Sensei_Email_Signup_Form::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the class and adds all filters and actions.
+	 *
+	 * @since 1.0.0
+	 */
+	public function init() {
+		// Add actions for displaying the email signup modal.
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_styles' ] );
+		add_action( 'admin_footer', [ $this, 'output_modal' ] );
+	}
+
+	/**
+	 * Enqueue the required JS assets for the modal dialog.
+	 *
+	 * @access private
+	 */
+	public function enqueue_scripts() {
+		// Register jQuery Modal JS.
+		wp_register_script(
+			'jquery-modal',
+			'https://cdnjs.cloudflare.com/ajax/libs/jquery-modal/0.9.1/jquery.modal.min.js',
+			false,
```

Comment:
Seems that jQuery modal would be dependent on jQuery.

Reply:
Good catch! https://github.com/Automattic/sensei/pull/2450/commits/ed6ca6b2087f7dcca9194936c918f7b40dc9cd95